packages/skeleton-svelte/src/lib/components/Switch/types.ts

```diff
@@ -82,4 +82,7 @@ export interface SwitchProps extends Omit<zagSwitch.Context, 'id'> {
 	inactiveChild?: Snippet;
 	/** The active state snippet. */
 	activeChild?: Snippet;
+	// Events ---
+	/** Triggers when the switch is toggled. */
+	onCheckedChange?: (value: boolean) => void;
```

Comment:
Why are we manually handling `onCheckedChange` instead of letting zag handle it? This would make the handler inconsistent with the rest of the components as they all delegate events to zag.

Reply:
Not super certain how Zag works but it seems that when passing a customer handler it doesn't work without triggering it like this.

Reply:
@Hugos68 this is why this one has been pending for a bit - I need to run through and verify if perhaps Svelte was doing it right and React doing it wrong. As much as possible we want to get out of the way and let Zag do it's thing.

Reply:
@phamduylong I will test it out, but other components don't have these issues so my thought is either it *does* work and you had something misconfigured or there is a bug in `zag` which has happened before as the zag Svelte package is relatively new.

Reply:
Ok after doing some tests I've discovered 2 things:
1. `onCheckedChange` works fine, so this is not something we need to touch.
2. `bind:checked` however currently does *not* update whenever the switch is checked or unchecked, so this will need to be fixed.

Reply:
@Hugos68 I'll give it a look

Reply:
@phamduylong FYI I've just fixed a similar issue in #2970, if you want to use that as a reference.

Reply:
@phamduylong @Hugos68 I've merged Hugo's change. If you want to update your branch with the latest changes, this is relevant in two ways:

1. You can implement Hugo's suggested change
2. We merged #2968 today, which enables the ability to add the test case Hugo suggested.

I'd welcome both. I'll also keep Hugo as a reviewer. He might be able to review before I can (which will be Monday).

Reply:
I rewrote the whole thing and added a test case for it. Hopefully behaves as it should now.